### PR TITLE
Fix GUI RunDialog not expanding details header

### LINKED ISSUE #6995: Max memory tab not expanding without selecting realization in gui
**Describe the bug**
The max memory tab should expand to fill the gap, or maybe it is better if all the tabs expand(?).
All tabs expand when a realization is specified, but it should probably not be showing before then anyways. 
Should be easily reproducible locally.

**Screenshots**
image
**Environment**
 - OS: macos
 - ERT/Komodo release: main
 - Python version: 3.10
 - Remote/HPC execution involved: no

**Additional context**

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #6995

**Approach**
The commit in this PR
 fixes an issue where the RunDialog show details section
would not expand the header to fill the width, without having a
realization set.

(Screenshot of new behavior in GUI if applicable)
image

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/gui/model/job_list.py [hunk 1]
```diff
@@ -81,22 +80,16 @@ def setSourceModel(self, sourceModel: QAbstractItemModel) -> None:
     @staticmethod
     def headerData(section: int, orientation: Qt.Orientation, role: Qt.UserRole) -> Any:
         if role != Qt.DisplayRole:
-            return QVariant()
+            return None
```
Comment (xjules) [inline src/ert/gui/model/job_list.py:83]: Is this related to the issue?

Comment (jonathan-eq) [reply src/ert/gui/model/job_list.py:83]: We made it more readable while trying to find the problem. Should I have it in a separate commit or drop it?

**Review by xjules — APPROVED** [review]
Very good! :rocket:
